src/v/archival/ntp_archiver_service.cc

```diff
@@ -464,6 +499,27 @@ ss::future<bool> ntp_archiver::sync_for_tests() {
     co_return false;
 }
 
+ss::future<std::error_code> ntp_archiver::process_anomalies(
+  cloud_storage::scrub_status status, cloud_storage::anomalies detected) {
+    // If there's ongoing housekeeping job, let it finish first.
+    auto units = co_await ss::get_units(_mutex, 1, _as);
+
+    auto sync_timeout = config::shard_local_cfg()
+                          .cloud_storage_metadata_sync_timeout_ms.value();
+    auto deadline = ss::lowres_clock::now() + sync_timeout;
+
+    auto error = co_await _parent.archival_meta_stm()->process_anomalies(
```

Comment:
nit: might be worth throwing in a check for `can_update_archival_metadata()`, though I guess I'm not too worried about this causing issues

Reply:
I'm not too worried about this tbh. If leadership moves away from the shard, we'll know via the abort source. If the term changes, it doesn't impact the scrubber.